client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx

```diff
@@ -74,20 +98,43 @@ export default function AssignLicenseForm( {
 						className="assign-license-form__site-card-radio"
 						label={ site.domain }
 						name="site_select"
-						disabled={ isLoading }
+						disabled={ ! isReady }
 						checked={ selectedSite?.ID === site.ID }
 					/>
 				</Card>
 			);
 		}
 	} );
 
-	const onAssignLater = () => {
+	const onClickAssignLater = useCallback( () => {
 		if ( licenseKeysArray.length > 1 ) {
 			page.redirect( '/partner-portal/licenses' );
 		}
+
+		const licenseKey = licenseKeysArray[ 0 ];
 		page.redirect( addQueryArgs( { highlight: licenseKey }, '/partner-portal/licenses' ) );
-	};
+	}, [ licenseKeysArray ] );
+
+	const onClickAssignLicenses = useCallback( async () => {
+		dispatch(
+			recordTracksEvent( 'calypso_partner_portal_assign_multiple_licenses_submit', {
+				products: licenseKeysArray.join( ',' ),
+				selected_site: selectedSite?.ID,
+			} )
+		);
+
+		const assignLicensesResult = await assignLicensesToSite( licenseKeysArray );
+
+		dispatch( resetSite() );
+		dispatch( setPurchasedLicense( assignLicensesResult ) );
+
+		const fromDashboard = getQueryArg( window.location.href, 'source' ) === 'dashboard';
+		if ( fromDashboard ) {
+			return page.redirect( '/dashboard' );
+		}
+
+		return page.redirect( partnerPortalBasePath( '/licenses' ) );
```

Comment:
Yeah! Generally my goal here was to separate the main concern of `useIssueAndAssignLicenses` (i.e., issuing and assigning licenses) from actions that take place visually in the browser, actions that modify the app state, or actions that record events for analytics. My desired end result was that the hook only handles the issuing and assigning of licenses, no more and no less.

I wasn't able to extract _every_ instance of these things because of how tightly coupled things were, but it's a big step in that direction.